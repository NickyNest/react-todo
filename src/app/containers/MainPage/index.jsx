import React, {Component} from 'react';
import Logo from 'components/Logo';
import './index.css';

class MainPage extends Component {
    constructor(props) {
        super(props);
        this.state = {
            pathToFile: 'src/containers/MainPage/index.jsx',
            counter: 0
        };
    }

    componentDidMount() {
        setInterval(this.increaseCounter, 1000);
    }

    increaseCounter = () => {
        this.setState({
            counter: this.state.counter + 1
        });
    }

    render() {
        const {pathToFile, counter} = this.state;
        const title = 'Welcome to React';
        return (
            <div className='App'>
                <div className='App-header'>
                    <Logo className='App-logo' alt='logo' />
                    <h2>
                        {title}
                    </h2>
                </div>
                <p className='App-intro'>
                    {'To get started, edit '}
                    <code>{pathToFile}</code>
                    {' and save to reload.'}
                </p>
                <p className='App-intro'>
                    {`Infinity increasing counter: ${counter}`}
                </p>
            </div>
        );
    }
}

export default MainPage;
