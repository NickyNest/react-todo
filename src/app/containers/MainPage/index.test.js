/* eslint-disable react/jsx-filename-extension */

import React from 'react';
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import MainPage from './index';

configure({adapter: new Adapter()});

describe('MainPage', () => {
    it('MainPage match expected snapshot', () => {
        expect(shallow(<MainPage />)).toMatchSnapshot();
    });
});
