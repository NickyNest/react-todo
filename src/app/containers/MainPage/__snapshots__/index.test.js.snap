// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPage MainPage match expected snapshot 1`] = `
<div
  className="App"
>
  <div
    className="App-header"
  >
    <Logo
      alt="logo"
      className="App-logo"
    />
    <h2>
      Welcome to React
    </h2>
  </div>
  <p
    className="App-intro"
  >
    To get started, edit 
    <code>
      src/containers/MainPage/index.jsx
    </code>
     and save to reload.
  </p>
  <p
    className="App-intro"
  >
    Infinity increasing counter: 0
  </p>
</div>
`;
